lib/global_config.js

```diff
@@ -256,4 +257,21 @@ exports.getKeepAlive = function () {
   return keepAlive;
 };
 
+let credentialManager = createDefaultCredentialManager();
 
+exports.setCustomCredentialManager = function (customCredentialManager) {
+  Errors.checkArgumentValid(Util.checkValidCustomCredentialManager(customCredentialManager),
+    ErrorCodes.ERR_GLOBAL_CONFIGURE_INVALID_CUSTOM_CREDENTIAL_MANAGER);
+  
+  credentialManager =  customCredentialManager;
+  Logger.getInstance().info('Custom credential manager is set by a user.');
+};
+
+exports.getCredentialManager = function () { 
+  return credentialManager;  
+};  
+
+function createDefaultCredentialManager() {
+  const JsonCrednetialManager = require('./authentication/secure_storage/json_credential_manager');
+  return new JsonCrednetialManager();
```

Comment:
for MFA tokens we will use the same class? should we pass the directory here?

Reply:
Yes, the original idea is to use the same file. I am not sure whether we need to create a different credential file or not.
Do you think that separating MFA tokens and ID Tokens is better? I agree with this.

Reply:
I think it could be the same file. If you see possibility to any conflict we can add type to generated key.

Reply:
No, there is no conflict because the key is generated based on the combination of the host, username, and token name (ID_TOKEN or MFA_TOKEN). The same file is also okay.